src/plat/pc99/machine/ioapic.c

```diff
@@ -151,16 +159,31 @@ void ioapic_map_pin_to_vector(word_t ioapic, word_t pin, word_t level,
 
     index = ioapic * IOAPIC_IRQ_LINES + pin;
     ioapic_write(ioapic, IOAPIC_REGSEL, IOREDTBL_HIGH(pin));
-    ioredtbl_high = ioapic_read(ioapic, IOAPIC_WINDOW) & MASK(IOREDTBL_HIGH_RESERVED_BITS);
+    ioredtbl_high = ioapic_read(ioapic, IOAPIC_WINDOW) & MASK(IOREDTBL_HIGH_DESTINATION_BITS);
+#ifdef CONFIG_IOMMU
+    word_t entry;
+    if (!vtd_create_irte_ioapic(vector, level, ioapic_target_cpu, &entry)) {
+        printf("IOAPIC: failed to create IRTE for vector %lu.", vector);
+        return;
+    }
```

Comment:
The `&entry` takes the address of a local variable. This is [not allowed in the kernel](https://docs.sel4.systems/processes/style-guide.html#verification-requirements).